howto.tex

```diff
@@ -0,0 +1,300 @@
+
+Create a conda environment based on the \textbf{docsteady} conda package:
+
+\texttt{conda create --name docsteady-env docsteady -c lsst-dm}
+
+Ensure that the conda configuration file  \texttt{.condarc} does not include the conda-forge channel.
+
+To use docsteady, activate the environment as follows:
+
+\texttt{conda activate docsteady-env}
+
+This environment will provide all dependencies that are required to run docsteady.
+
+It is recommended to use the provided conda environment also for development activities, see section \ref{sec:development}.
+
+
+\section{Execution}
+
+It is recommended to provide the credentials seting up the following environment variables:
+
+\begin{itemize}
+\item \texttt{export JIRA\_USER=<jira-username>}
+\item \texttt{export JIRA\_PASSWORD=<password>}
+\end{itemize}
+
+otherwise, it is required to specify them from the command line using the \texttt{--username} and \texttt{--password} options.
+In case credential options are omitted and no environment variables are defined, username and password will be prompted interactively.
+
+Personal Jira credentials can be used. 
+For CI purposes, a general set of credentials are available, as specified in section \ref{sec:auth}.
+
+In order to execute any of the docsteady commands described in the following subsections, a conda environment providing docsteady shall be activated, as described in section \ref{sec:install}.
+
+Use \texttt{--help} to get the list of available options for each of the docsteady commands.
+In addition, existing documents generated using docsteady, have a \texttt{.docugen} file with the command to be executed in CI, which can be used as an example.
+
+
+
+\subsection{Test Specification Generation}
+
+Test specifications are extracted using REST API.
+All tests cases included in a TM4J folder, including subfolders, are rendered in the same extraction.
+The folder organization in Jira should correspond to the major subsystem components, and modeled in MagicDraw.
+
+The syntax to extract a test specification is the following:
+
+\texttt{docsteady generate-spec "</tm4j\_folder>" jira\_docugen.tex}
+
+where \texttt{</tm4j\_folder>} shall be replaced by the exact folder where the test cases are defined in the Jira \texttt{Test Cases} tab.
+For example, the command to extract the DM Acceptance test specification, LDM-639, is the following:
+
+\texttt{docsteady generate-spec "/Data Management/Acceptance|LDM-639" [jira\_docugen.tex]}
+
+Note that:
+\begin{itemize}
+\item the output file \textit{jira\_docugen.tex} is optional in the execution of docsteady, but required in this context in order to include the extracted information in a \LaTeX~document (i.e. LDM-639.tex). If omitted, the docsteady  output will just be printed in the terminal;
+\item the folder name in Jira includes at the end the test specification document handle. This is a best practice to use when organizing test cases in Jira, it helps to orientate in the folder structure;
```

Comment:
```suggestion
\item the folder name in Jira includes at the end the test specification document handle. This is a best practice to use when organizing test cases in Jira since it helps to orient the user in the folder structure;
```